src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final Random RANDOM = new Random();
+    private static final ColorSupplier COLOR_SUPPLIER = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Type figures = Type.values()[new Random().nextInt(Type.values().length)];
+        switch (figures) {
+            case SQUARE: {
+                Square square = new Square(RANDOM.nextInt(100),

```

Comment:
Looks like creating `square` variable is redundant here. Same about other figures, we can just return new instance, not storing it in a variable.